### PR TITLE
Improve exception message on corrupted repository

### PR DESCRIPTION
**Pull request by kingherc**
After #89130 was solved, realized that this exception message can be now improved to mention to re-create the repository with the same settings.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -2023,7 +2023,7 @@ private RepositoryException corruptedStateException(@Nullable Exception cause, @
                 + "expected state. This is likely the result of either concurrently modifying the contents of the "
                 + "repository by a process other than this cluster or an issue with the repository's underlying storage. "
                 + "The repository has been disabled to prevent corrupting its contents. To re-enable it "
-                + "and continue using it please remove the repository from the cluster and add it again to make "
+                + "and continue using it please recreate the repository with the same settings to make "
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:2026]: May be it is better not to mention settings?

**Review by DaveCTurner — COMMENTED** [review]
I worry that `recreate the repository` isn't meaningful to users (or may be misinterpreted as requiring the repository to be deleted first). I also think there's other room for improvement in this message. How about something like this?

```java
    private RepositoryException corruptedStateException(@Nullable Exception cause, @Nullable Tuple<Long, String> previousWriterInfo) {
        return new RepositoryException(metadata.name(), """
            The repository has been disabled to prevent data corruption because its contents were found not to match its expected state. 
            This is either because something other than this cluster modified the repository contents, or because the repository's 
            underlying storage behaves incorrectly. To re-enable this repository, first ensure that this cluster has exclusive write 
            access to it, and then use the Update Snapshot Repository API to re-register the repository with this cluster.
            """ + previousWriterMessage(previousWriterInfo), cause);
    }
```

**Comment by kingherc** [comment]
Hi @DaveCTurner and @idegtiarenko , thanks for the meaningful feedback! Do you think we can mention the documentation URL (https://www.elastic.co/guide/en/elasticsearch/reference/8.5/add-repository.html), which I modified in the previous PR, in the exception that covers all cases of how to update the corrupt repository (e.g., on premise but also on cloud)?

I see this is happening already in a couple of places, e.g., at [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/SecurityImplicitBehaviorBootstrapCheck.java#L39-L48) and [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/tool/BaseRunAsSuperuserCommand.java#L252-L257). But it doesn't seem to be a popular approach. However, I would prefer it in this situation, because the reference provides more details on how to handle the corrupt repository.

What do you think?

**Comment by DaveCTurner** [comment]
👍 yes nice idea.

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 2]
```diff
@@ -2019,12 +2019,15 @@ private void cacheRepositoryData(RepositoryData repositoryData, Version version)
...
-                + "the cluster recover the known state of the repository from its physical contents."
+            "The repository has been disabled to prevent data corruption because its contents were found not to match its expected state. "
+                + "This is either because something other than this cluster modified the repository contents, or because the repository's "
+                + "underlying storage behaves incorrectly. To re-enable this repository, first ensure that this cluster has exclusive "
+                + "write access to it, and then use the Update Snapshot Repository API to re-register the repository with this cluster. "
+                + "See https://www.elastic.co/guide/en/elasticsearch/reference/"
+                + Version.CURRENT.major
+                + "."
+                + Version.CURRENT.minor
+                + "/add-repository.html for detailed instructions on how to re-register the repository."
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: Oh sorry, https://www.elastic.co/guide/en/elasticsearch/reference/current/add-repository.html does indeed give detailed instructions. I thought it was just the API docs.

In that case we can drop the bit about `use the Update Snapshot Repository API`

[the remark this answers is not in the document]
Comment (kingherc) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: @DaveCTurner Sure, just did.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by kingherc** [comment]
@elasticmachine please run elasticsearch-ci/bwc

**Comment by kingherc** [comment]
@elasticmachine please run elasticsearch-ci/part-1

**Comment by kingherc** [comment]
@elasticmachine please run elasticsearch-ci/part-2

**Comment by kingherc** [comment]
@elasticmachine please run elasticsearch-ci/part-3

**Pull request merged**